### PR TITLE
Demo

### PR DESCRIPTION
**Pull request by iarry**
#### What does this PR do?

Adds a demo and mentions it in the README. Also removes the `npm-debug.log`.
#### Where should the reviewer start?

Open `demo/index.html` in a browser - no `npm install` or other build steps necessary.
#### How should this be manually tested?

You can restart the demo with the `Reload` button. The button should be disabled while loading, and the loader should finish in 1500ms.
#### Any background context you want to provide?

React, ReactDOM, Spin.JS, and a Babel compiler all reference a CDN to avoid any build process for the demo. The Babel compiler was necessary for JSX (wanted to stay consistent with the README).
#### What are the relevant issues (if any)?
#23
#### Screenshots (if appropriate)

See `demo/demo.gif`
#### Obligatory GIF

See `demo/demo.gif`

### CODE REVIEW

**Comment by oyeanuj** [comment]
@mikekauffman @mhuggins Any chance this can be merged? It would be great to see a demo of the library easily.

File: demo/index.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    const Demo = React.createClass({
+      getInitialState: function () {
+        return { loaded: true, data: null };
+      },
+      componentDidMount: function() {
+        this.simulateLoad();
+      },
+      simulateLoad: function() {
+        this.setState({ loaded: false, data: null });
```
Comment (mhuggins) [inline demo/index.html]: I would argue that the initial state should set loaded to false, and this line can go away.

File: demo/index.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        // Simulated a response callback/promise/etc.
+        setTimeout(function() {
+          this.onSuccess('Hello World!');
+        }.bind(this), 1500);
+      },
+      onSuccess: function (data) {
+        this.setState({ loaded: true, data: data });
+      },
+
+      onError: function (err) {
```
Comment (mhuggins) [inline demo/index.html]: This isn't being bound anywhere, so I'd consider just removing it from the example altogether.

File: demo/index.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        // error handling goes here
+      },
+      render: function () {
+        return (
+          <div className="demo">
+            <Loader loaded={ this.state.loaded }>
+              <div className="card">{ this.state.data }</div>
+            </Loader>
+            <button onClick={ this.simulateLoad } disabled={ this.state.loaded ? false : true }> Reload </button>
```
Comment (mhuggins) [inline demo/index.html]: :neckbeard: `disabled={!this.state.loaded}`

**Comment by mhuggins** [comment]
:cool: Thanks for the contribution!  I don't have access to the repo anymore, but other than my few minor pieces of feedback, I'd say this looks pretty good. :)

**Comment by iarry** [comment]
Sounds good, I can make these changes to the PR tomorrow morning.

**Comment by kclowes** [comment]
Thank you for this PR! :boom:

**Pull request merged**